src/rust/ide/src/view/node_editor.rs

```diff
@@ -262,19 +268,22 @@ impl GraphEditorIntegratedWithControllerModel {
 
 // === Updating Graph View ===
 
-impl GraphEditorIntegratedWithControllerModel {
+impl GraphEditorIntegratedWithModelModel {
     /// Reload whole displayed content to be up to date with module state.
     pub fn refresh_graph_view(&self) -> FallibleResult<()> {
+        info!(self.logger, "Refreshing graph view");
         use controller::graph::Connections;
-        let Connections{trees,connections} = self.controller.graph.connections()?;
+        let Connections{trees,connections} = self.model.graph().connections()?;
         self.refresh_node_views(trees)?;
         self.refresh_connection_views(connections)?;
         Ok(())
     }
 
     fn refresh_node_views
     (&self, mut trees:HashMap<double_representation::node::Id,NodeTrees>) -> FallibleResult<()> {
-        let nodes = self.controller.graph.nodes()?;
+        debug!(self.logger, "Updating nodes for {self.model.graph():?}");
+        let nodes = self.model.graph().nodes()?;
+        debug!(self.logger, "Updated nodes {nodes:?}");
```

Comment:
debug messages should be sentences (missing dots)

Reply:
fixed